src/codegen.cpp

```diff
@@ -8123,8 +8123,7 @@ void jl_compile_workqueue(
             else {
                 // Reinfer the function. The JIT came along and removed the inferred
                 // method body. See #34993
-                if (policy != CompilationPolicy::Default &&
-                    codeinst->inferred && codeinst->inferred == jl_nothing) {
+                if (policy != CompilationPolicy::Default && jl_atomic_load_relaxed(&codeinst->inferred) == jl_nothing) {
```

Comment:
Extract into a temporary value `jl_value_t *inferred = jl_atomic_load_relaxed(&codeinst->inferred)` and insert explicit `NULL` check `if (inferred && inferred == jl_nothing)` following what we do for the other parts?

Reply:
I'm not seeing what the explicit NULL check contributes to the if statement. If it is null, it's not going to be == to jl_nothing anyways, and if it's equal to jl_nothing we know it's non-null.

Reply:
Yeah I agree, I wonder if we want to simplify this pattern found in other parts, e.g. 
- https://github.com/JuliaLang/julia/blob/1f421484e0711804ad81083218a0818eb62dc4f7/src/precompile.c#L339
- https://github.com/JuliaLang/julia/blob/1f421484e0711804ad81083218a0818eb62dc4f7/src/staticdata.c#L1710